generators/client/templates/angular/src/main/webapp/app/app-routing.module.ts.ejs

```diff
@@ -53,7 +53,7 @@ const LAYOUT_ROUTES = [navbarRoute, ...errorRoute];
         <%_ } _%>
         ...LAYOUT_ROUTES,
       ],
-      { enableTracing: DEBUG_INFO_ENABLED }
+      { enableTracing: DEBUG_INFO_ENABLED, relativeLinkResolution: 'legacy' }
```

Comment:
Do we have code which forces us to use `relativeLinkResolution: 'legacy'`?

If yes then can we changes those to use Angular 11 default `relativeLinkResolution: 'corrected'`?

Reply:
I don't know, it's this command:`ng update @angular/cli @angular/core` to migrate which updates this file

Reply:
I think we should migrate from `'legacy'` to `'corrected'`.  
Tests are failing because `DatePipe` has stricter type checking in Angular 11 and code needs to be changed.  
@qmonmert are you planning to work on this PR to get CI green?

Reply:
yes I'm working on it tomorrow, I have already started (but not push on this PR)

Reply:
@kaidohallik do you have an idea to fix `DatePipe` with `Dayjs` ?

Reply:
`DayJs`  has https://day.js.org/docs/en/display/format, maybe use this instead of `DatePipe`? If using methods in HTML then it's also good idea to switch to `changeDetection: ChangeDetectionStrategy.OnPush`, to avoid bad performance.

Reply:
The pipes are pure by default so for me the OnPush strategy is not mandatory, the pipe code will be executed only when the input changes even if the Component is not using the OnPush strategy

Reply:
Pipes are replaced by methods, so `OnPush` is needed for good performance.

Reply:
when I use `changeDetection: ChangeDetectionStrategy.OnPush` on my component, the list of entities does not display

Reply:
On data load use `this.changeDetector.markForCheck();` as this is done in `metrics.component.ts`.

Reply:
yes it works

Reply:
Maybe we can use custom date pipes using dayjs instead of the Angular default ones ? This way we can remove the OnPush and provide our users some dayJS pipes. 
For me formatting an input for the display is why pipes are made for., WDYT ?

That can be done on a separate PR

Reply:
For `DayJs` pipes is library available: https://github.com/ljmerza/ngx-dayjs - so if you think we should go for pipes then first need discuss, use this library or not.

Reply:
The lib doesn't look maintained so I won't go for it, I would vote for putting two custom pipes in our shared folder

Reply:
Agree with custom pipes in our shared folder, there is already `duration.pipe` for `DayJs`. So I suggest to create `shared/dayjs` subfolder, move `duration.pipe` there and add additional pipes.

Reply:
ok let's do this

Reply:
Maybe `shared/dates` WDYT ? 😄  For me that should be an abstract thing so we can easily change our date lib in the future

Reply:
Yes, this looks better, but maybe not in plural, but `shared/date` like other folders are named?